File: JavaScript/Sorting Algorithms/insertionSort.js
function insertionSort(arr) {
    var currentVal;
    for (var i = 1; i < arr.length; i++) {
        currentVal = arr[i];
        for (var j = i - 1; j >= 0 && arr[j] > currentVal; j--){
            arr[j+1] = arr[j]
        }
        arr[j+1] = currentVal;
    }
    return arr;
}

console.log(insertionSort([100,50,25,10,5, 16]))
console.log(insertionSort([2,3,4,5,6]))

//100
//50 25 10 5 16

File: Python/Sorting Algorithms/insertionSort.py
def insertionSort(arr):
    for i in range(1, len(arr)):
        currentVal = arr[i]
        j = i - 1
        while j >= 0 and currentVal < arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = currentVal 
    return arr


print(insertionSort([100,50,25,10,5, 16]))
print(insertionSort([2,3,4,5,6]))

